### DISCUSSION TITLE
Síndrome do Impostor

### DISCUSSION THREAD

**Discussion by rubenmarcus** [frontendbr/forum#372] [Discussão]
Galera, estava lendo esse artigo na Smashing Magazine

https://www.smashingmagazine.com/2016/11/not-an-imposter-fighting-front-end-fatigue/

O que pensam sobre? Tambem se sentem, ou já se sentiram assim? (se puder falem como fazem para aprender mais e mais e mais.. e melhor)

Abraços

**Comment by felipefialho** [comment]
Como em tudo na vida, costumo ser pragmático.

Quando vejo uma matéria ou nova feature que parece interessante, primeiro passo o olho e filtro: "Isso vai ser útil no meu workflow a curto ou médio prazo?".

Se não for, eu passo e leio coisas que posso aplicar agora ou muito em breve. Na verdade não me importo nem um pouco de estar desalinhado com os hypes do momento (e toda hora tem um).

Sério, trabalhamos com o cérebro, e a sanidade mental precisa estar boa para que a gente possa produzir e ser útil no trabalho/comunidade.

O mais importante não é a quantidade de conhecimento que acumula, mas sim a qualidade dele (para sua vida/profissão).

**Comment by talitapagani** [comment]
Nossa, passo por isso direto!

Ainda mais por ter ficado quase 2 anos na área de gestão de projeto e ter perdido o boom de alguns frameworks, sempre me sinto atrás, nem tão competente assim, aquém do que é esperado para minha carreira, um zero à esquerda comparada aos meus pares. Isso faz com que a gente se sabote muito e se valorize menos.

Tem o fator comparação também: sempre achamos que o colega do lado programa melhor, é mais rápido, sabe muito mais, etc. O grande exercício é pararmos de pensar que a grama do vizinho é sempre mais verde, valorizar nossos pontos fortes e converter os pontos que consideramos fracos para algo bom. Não é fácil não, rs.

Eu não tenho conhecimento aplicado da maioria dos frameworks requeridos pelas vagas. Mas tenho conhecimentos de desenvolvimento que são tão valiosos quanto.

E, fazendo coro ao @LFeh, o importante é a qualidade do que se sabe. Uma coisa que sempre digo: aprenda conceitos em profundidade antes de ferramentas. Conceito tem uma vida útil muito maior que ferramentas e pode ser "instanciado" em diferentes contextos. Ferramentas se tornam obsoletas com facilidade, mas também são mais fáceis de aprender.

Então é isso, aprenda o que for necessário, esteja informado sobre o que está rolando, mas não se prenda a saber tudo sobre tudo ;)

**Comment by felipefialho** [comment]
Fazendo duplo coro a @talitapagani agora, no começo do ano passado escrevi [esse artigo](http://www.felipefialho.com/blog/2015/meu-dia-a-dia-como-dev-frontend).

E no final dele escrevi:

> Importante lembrar que ferramentas, aplicativos e todas essas maravilhas modernas, são apenas facilitadores para que possamos entregar o melhor trabalho possível. Se apegue mais ao conceito, e menos nas ferramentas, porque normalmente, elas são passageiras.

Continuo concordando comigo mesmo.

**Comment by rubenmarcus** [comment]
sim o artigo resume muito o que a @talitapagani  falou. 

essa imagem por exemplo :

![Job](https://www.smashingmagazine.com/wp-content/uploads/2016/09/jobs-edited-500-opt.png)


Mas uma coisa que li nos comentários a galera falando é exatamente sobre aprendizado.

Muito do que você faz para sua empresa é dificil você ter essa lista inteira de conhecimento , e também não é raro ficar mais de 10h por dia trabalhando com coisas repetitivas

HTML (writing clean, semantic markup)
CSS (Modular, scalable)
CSS methodologies (BEM, SMACSS, OOCSS)
CSS preprocessors (something like LESS, SCSS, PostCSS)
Modern CSS (Flexbox, Grid)
JS
Modern JS (ES6, Typescript)
JS frameworks (Angular, React, Vue [insert latest here]
JS methodologies (Functional programming, OOP)
JS libraries (Immutable, Ramda, Lodash)
Responsive Design principals
Testing (TDD)
Testing frameworks (Jasmine, Karma)
SVG
WebGL
Animation techniques
Accessibility
Usability
Performance
Build tools (Grunt, Gulp, NPM Scripts)
Asset Bundlers (WebPack, Browserify)
NPM ecosystem
Knowledge of different browser quirks
Agile methodologies
Version Control (Usually Git)
Visual Design fundamentals
Soft skills, time management
A basic understanding of whatever back-end language is being used
And on top of that you’re either dabbling with or looking towards things like:
Service workers
Progressive Web Apps (PWA)
Web Components



como não sacrificar tanto o tempo livre com aprendizado? Porque pode até parecer imoral,mas temos que sair, nos divertir, ter hobbies, passar tempo com a familia e etc..  a vida não é só a carreira 


pra você ter os conhecimentos mais ou menos, até se consegue, mas avançado nisso tudo é bem complicado, tem que ter muitos anos de experiência ou ser um gênio,porque é muita informação

No meu caso, eu tenho muito mais interesse em bibliotecas visuais ( Processing, d3) e animação css e também UI/UX, e sei que não tem tanto mercado quanto recorte de layout,responsividade ( o que faço basicamente )...

**Comment by diegoslva** [comment]
Por experiência própria parei de seguir os hypes do momento. Na gana de querer saber e aprender tudo que estava em pauta naquele momento, acabei me perdendo e sabotando a minha carreira. Passei cerca de dois anos entender isso. Estava sempre correndo atrás e não chegando a lugar algum. 

Hoje tenho mais clareza da onde quero chegar e o que realmente importa. Sei que tenho que aprender e aprender bem o inglês, conceitos e fundamentos base,  e ai sim poder evoluir em novas habilidades conforme a necessidade. Talvez assim posso acompanhar,  mesmo que de longe, novas tecnologias sem cair na loucura de querer sair usando tudo apenas por que outras pessoas estão usando ou por medo de ficar para traz.  Claro que temos que evoluir e acompanhar as novidades para não ficar estagnado. 

Mas Para cair de cabeça numa tecnologia nova ou algo no gênero, contexto é a palavra chave. Se algo me serve e resolve o meu problema, faz sentido usar, se não, deixo dentro de uma caixinha para um dia caso precise saber que aquela parada vai resolver o meu problema. 

Essa nova forma de pensar tem me ajudado bastante, me mantenho mais motivado e sem duvida aprendi muito mais  em 2 meses do que em dois anos.

**Comment by AlexAvila** [comment]
Ainda sobre o assunto, quero recomendar um episódio do podcast TravandLos sobre síndrome do impostor. http://www.travandlos.com/69. Não fala sobre o aspecto técnico de Front-end, mas fala sobre o aspecto psicológico do tema.

**Comment by viniciusdocanto** [comment]
Eu tento fugir o máximo do buzz do momento, sempre me espelho nas vagas de emprego olhando o geral do que o pessoal ta pedindo, vejo uma coisa que a maioria das vagas ta pedindo e dou uma olhada vejo se é interessante/útil pro meu dia dia e decido a partir dai estudar ou não isso.

**Comment by danjesus** [comment]
Esse tópico é bem interessante, estava falando isso com um amigo no trabalho hoje mesmo sobre, já me senti muito mal por pensar que estava bem abaixo em termos de conhecimento que muitos na comunidade ou que palestram por ai, a um ponto de isso me fazer mal e me desmotivar.

Sempre achava que todos estavam bem melhor, eram mais rápidos assim como a @talitapagani falou mais acima, afinal nossa profissão é muito dinâmica e tudo muda em questão de horas, acho que um framework novo é lançado a cada 3 horas rs💃 .

Mas isso mudou em 2015 eu comecei a trabalhar numa startup bem legal(Tripda que infelizmente não existe mais) e com pessoas sensacionais que me fez mudar o pensamento, dentro daquele time pude perceber o quanto cada desenvolvedor é diferente e quanto cada um sabe coisas diferentes um do outro, eu parei de olhar para a grama do vizinho e comecei a entender que cada um tem seu ritmo de aprendizado, qualidades e habilidades.

O que eu fiz e faço pra manter essa sensação sobre controle é estudar fundamentos e não mais a Hype, outros acontecimentos também fizeram eu mudar bastante minha visão com relação a isso, como o nascimento do filho.

Além disso tudo, acho que todos nós temos uma grande necessidade de sermos reconhecidos pelas nossas qualidades e isso acaba gerando uma pressão interna muito forte, nesse ponto cheguei a conclusão que não tem ponto de vista certo ou errado sobre um problema ou código, tem a implementação que é a visão da pessoa que codificou, ela pode ser mais clara que a sua mas isso não invalida sua hipótese, código que funciona  é uma excelente medida de progresso e vale tanto quanto desde que você se dedique com todo o coração no que faz.

**Comment by fabriciofmsilva** [comment]
Acredito que essa síndrome do impostor acontece por alguns motivos: primeiro, somos seres sociais. Segundo que culturalmente aprendemos a validar aquilo que pensamos/sabemos/fazemos com as pessoas de nossos círculos sociais. Acontece que com a Internet podemos participar de qualquer círculo, mesmo que como espectadores. A partir dessa premissa, queremos cada vez mais nos comparar/validar com as pessoas que admiramos.

Queremos estar na moda, na crista da onda, queremos ser os melhores no que fazemos. E infelizmente usamos a comparação com os outros como parâmetro de crescimento. De um tempo para cá eu venho tentado apenas me comprar com o que eu fui ontem. Em melhorar a passos curtos, aprender devagar, e melhorar sempre (melhoria contínua).

Como o pessoal já falou anteriormente, é comum cairmos no pensamento que somos 'piores' que outros. Quando na verdade somos apenas diferentes. Não valorizamos as nossas experiências e qualidades. As dicas dadas também são as que eu procuro seguir, aprendendo os conceitos, fundamentos, criando uma base cada vez mais sólida. Acredito que só assim é possível criar um alicerce que se sustente e posteriormente sustente os pilares.

**Comment by willianjusten** [comment]
Fiz um post sobre isso há pouco tempo e abordei um pouquinho dessas coisas. Nem imaginava que já tinha até uma issue aqui sobre isso, irei referenciar lá =D

Btw, se alguém quiser ler o [post o link está aqui](https://willianjusten.com.br/sou-bom-o-suficiente/).

**Comment by hjdesigner** [comment]
Eu passo por isso toda vez que recebo uma vaga de emprego.

**Comment by douglasbastos** [comment]
Creio que isso acontece normalmente no início de carreira, depois de um tempo aprendemos a filtrar todas essas informações e investir somente o que enchergamos que é necessário. 

Escrevi sobre isso alguns meses atrás voltado para quem incia na área entitulei como a saga do programador iniciante, para quem quiser conferir. 

https://medium.com/@douglasbastosok/a-saga-do-programador-iniciante-61514c62df33

**Comment by talitapagani** [comment]
@douglasbastos infelizmente isso também acomete muito os profissionais de nível sênior. Acaba piorando com o tempo, porque aí pensamos que a gente não merece estar onde está e que a qualquer momento seremos "descobertos" como fraude. Tenho quase 12 anos de carreira e ainda passo por isso. É irracional, mas é um sentimento muito frequente até mesmo para os mais experientes 😢

**Comment by douglasbastos** [comment]
@talitapagani Então só tende a piorar na maioria dos casos.

Mas eu aprendi a lidar com isso, e coloquei na minha cabeça que sempre existirá alguém melhor que eu, como também vai existir alguém que não sabe tanto quando eu.

Atualmente sou Junior, mas tiro dúvida de git para um cara Sênior. E isso não que dizer que sou melhor que ele, apenas ele veio do mundo Microsoft e nunca tinha utilizado git.

Resumindo devemos nos cobrar menos, E não tentar aprender tudo de uma vez.

Um trecho que está no meu artigo que gosto bastante.

> O programador <s>iniciante</s> fica muito atormentado com a quantidade de tecnologia que existem e que devem aprender. Acaba se afobando e querendo dominar todas essas tecnologias de uma só vez, isso é humanamente impossível e demanda bastante tempo, muitos dos programadores de referência podem saber bastante sobre ferramenta x, mas y e z eles apenas sabem se virar. E lembre-se esses caras já estão a muito tempo fazendo o que você começou a fazer a pouco tempo.

**Comment by talitapagani** [comment]
@douglasbastos post muito bacana! Não é fácil lidar com isso, principalmente para quem ainda não sabem que tem. Faz pouco tempo que ouvi falar que tinha um nome e que mais gente passava por isso. É um exercício constante para combater.

**Comment by kelciocajueiro** [comment]
Certeza que tenho isso :)

**Comment by douglas-castilho** [comment]
Eu diria que um bom exercício é não se preocupar tanto com os hype que tem nessa área, temos que estar atentos, mas não focados só em ver coisas novas.
Outra coisa que eu acho muito importante, focar na base e não só tecnologicamente falando. Temos sempre que estar trocando experiências, repassando o que sabemos e absorvendo o conhecimento das outras pessoas.

**Comment by thulioph** [comment]
Comunidade conforta, abre os olhos e não te deixa só nunca. Esses dias vinha "sofrendo" por conta disso, lendo aqui você termina sabendo que não está só nessa..

Daniel Filho falou um pouco sobre esse assunto no [Foocast - Sindrome do Impostor](http://foocast.io/episode/2016/03/29/sindrome-do-impostor.html).

**Comment by andrewsfg** [comment]
Eu também tenho síndrome do impostor, mesmo estando há mais de 10 anos na área, assim como a @talitapagani descreveu, não tenho muito o que incrementar, só dizendo que essa frustração vira uma cobrança pessoal enorme que te coloca em dúvida muitas vezes sobre qual caminho seguir . A experiência que tenho com projetos são mais vistas se usei framework x, framework y e não os desafios, o esforço e o resultado alcançado. 

Nossa área muda muito rápido e parece que sempre estamos trabalhando loucamente 24 horas por dia. Essa sensação é complicada, eu tento muito priorizar a vida pessoal hoje e não só o trabalho, porque quando isso virá problema de saúde, a situação muda muito de lado.

E sim, a comunidade conforta. Obrigado pela opinião de vocês.

**Comment by felipefialho** [comment]
Complementando o post, li esse artigo faz alguns anos, mas é o melhor que já vi sobre o assunto:

http://qga.com.br/comportamento/jovem/2013/09/porque-os-jovens-profissionais-da-geracao-y-estao-infelizes

**Comment by jaffarc** [comment]
Eu posso ser um impostor para muitas empresas, pois não sei trabalhar com frameworks na parte do FrontEnd.  Atualmente trabalho  em um empresa que não  liga  para isso. 
Apesar  que nos ultimos 8 meses estou  apenas backEnd nodejs.

**Comment by tghelere** [comment]
Podcast do Dev na Estrada sobre este tema:
https://devnaestrada.com.br/2017/02/03/sindrome-do-impostor.html
